labelbox/schema/dataset.py

```diff
@@ -750,3 +755,35 @@ def _export(
         res = res[mutation_name]
         task_id = res["taskId"]
         return Task.get_task(self.client, task_id)
+
+    def upsert_data_rows(self, items: list[DataRowUpsertItem]) -> "Task":
+        chunk_size = 3
+        chunks = [
+            items[i:i + chunk_size] for i in range(0, len(items), chunk_size)
+        ]
+        manifest = {"source": "SDK", "item_count": 0, "chunk_uris": []}
+        for chunk in chunks:
+            manifest["chunk_uris"].append(self._create_descriptor_file(chunk))
+            manifest["item_count"] += len(chunk)
+
+        data = json.dumps(manifest).encode("utf-8")
+        manifest_uri = self.client.upload_data(data,
+                                               content_type="application/json",
+                                               filename="manifest.json")
+
+        dataset_param = "datasetId"
+        manifest_uri_param = "manifestUri"
+        query_str = """mutation UpsertDataRowsPyApi($%s: ID!, $%s: String!){
+                upsertDataRows(data:{datasetId: $%s, manifestUri: $%s}
+                ){ id createdAt updatedAt name status completionPercentage result errors type metadata } } """ % (
+            dataset_param, manifest_uri_param, dataset_param,
+            manifest_uri_param)
+
+        res = self.client.execute(query_str, {
+            dataset_param: self.uid,
```

Comment:
We should validate the dataset id set on each specification (if any) matches what is set here.